provisioner/src/lib.rs

```diff
@@ -470,7 +486,33 @@ impl Provisioner for ShuttleProvisioner {
                     .await?
             }
             DbType::AwsRds(AwsRds { engine }) => {
-                can_provision_rds?;
+                {
+                    let mut rr_client = self.rr_client.lock().await;
+
+                    let can_provision = match claim
+                        .can_provision_rds(&self.gateway_client, &mut *rr_client)
+                        .await
```

Comment:
It looks like the tracing context will be propagated in the clients so we can trace the whole flow from "provision_database" to r-r and gateway, but should we instrument this call so we can isolate its duration ? This function will also call the aws sdk, so it would be good to be able to separate the time it spends in that call and this one.

Reply:
Very good idea. I'll add that :+1: